src/main/java/net/minecraftforge/event/entity/living/LivingBurnBySunlightEvent.java

```diff
@@ -0,0 +1,36 @@
+package net.minecraftforge.event.entity.living;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraftforge.eventbus.api.Event.HasResult;
+
+/**
+ * LivingBurnBySunlightEvent is triggered when an entity is trying
+ * to burn (caused by sunlight) during {@link LivingEntity#livingTick()}
+ * through {@link net.minecraft.entity.Entity#setFire(int)}
+ *
+ * This event has a result. {@link HasResult}
+ * {@link Result#ALLOW} Burn the entity for custom duration
+ * {@link Result#DENY} Prevent the entity from burning
+ * {@link Result#DEFAULT} Burn the entity for original duration
+ */
+@HasResult
```

Comment:
Not quite sure if this should really have a result. If you call `setDuration`, you 100% want the custom duration, if you don't want the entity to burn set the duration to zero. Maybe cancelable would make sense, but even that is questionable

Reply:
I think cancelable does make sense, as setFire(0) wil do nothing.